src/editors/textEditor.js

```diff
@@ -18,413 +18,476 @@ import EventManager from './../eventManager';
 import { KEY_CODES } from './../helpers/unicode';
 import { stopPropagation, stopImmediatePropagation, isImmediatePropagationStopped } from './../helpers/dom/event';
 
-const TextEditor = BaseEditor.prototype.extend();
-
 /**
  * @private
  * @editor TextEditor
  * @class TextEditor
  * @dependencies autoResize
  */
-TextEditor.prototype.init = function() {
-  const that = this;
-  this.createElements();
-  this.eventManager = new EventManager(this);
-  this.bindEvents();
-  this.autoResize = autoResize();
-  this.holderZIndex = -1;
-
-  this.instance.addHook('afterDestroy', () => {
-    that.destroy();
-  });
-};
-
-TextEditor.prototype.prepare = function(row, col, prop, td, originalValue, cellProperties, ...args) {
-  const previousState = this.state;
-
-  BaseEditor.prototype.prepare.apply(this, [row, col, prop, td, originalValue, cellProperties, ...args]);
-
-  if (!cellProperties.readOnly) {
-    this.refreshDimensions(true);
-
-    const {
-      allowInvalid,
-      fragmentSelection,
-    } = cellProperties;
-
-    if (allowInvalid) {
-      this.TEXTAREA.value = ''; // Remove an empty space from texarea (added by copyPaste plugin to make copy/paste functionality work with IME)
-    }
-
-    if (previousState !== EditorState.FINISHED) {
-      this.hideEditableElement();
-    }
-
-    // @TODO: The fragmentSelection functionality is conflicted with IME. For this feature refocus has to
-    // be disabled (to make IME working).
-    const restoreFocus = !fragmentSelection;
+class TextEditor extends BaseEditor {
+  /**
```

Comment:
Missing constructor with defined all properties used in this editor (https://github.com/handsontable/handsontable/pull/5628/files#diff-6c12546315d5a1a24cf343b03ff3ed89R155).